contracts/libraries/Assessment/AssessmentVoteLib.sol

```diff
@@ -0,0 +1,176 @@
+  //  views for total rewards and withdrawable rewards
+  function withdrawReward (
+    IAssessment.Configuration calldata CONFIG,
+    INXMToken nxm,
+    address user,
+    uint104 untilIndex,
+    mapping(address => IAssessment.Stake) storage stakeOf,
+    mapping(address => IAssessment.Vote[]) storage votesOf,
+    IAssessment.Claim[] storage claims,
+    IAssessment.Incident[] storage incidents
+  ) external returns (uint rewardToWithdraw, uint104 withdrawUntilIndex) {
+    IAssessment.Stake memory stake = stakeOf[user];
+    {
+      uint voteCount = votesOf[user].length;
+      withdrawUntilIndex = untilIndex > 0 ? untilIndex : uint104(voteCount);
+      require(
+        untilIndex <= voteCount,
+        "Vote count is smaller that the provided untilIndex"
+      );
+      require(stake.rewardsWithdrawnUntilIndex < voteCount, "No withdrawable rewards");
+    }
+
+    uint totalReward;
+    for (uint i = stake.rewardsWithdrawnUntilIndex; i < withdrawUntilIndex; i++) {
+      IAssessment.Vote memory vote = votesOf[user][i];
+      require(
+        block.timestamp > AssessmentUtilsLib._getVoteLockupEndDate(CONFIG, vote),
+        "Cannot withdraw rewards from votes which are in lockup period"
+      );
+      IAssessment.Poll memory poll =
+        IAssessment.EventType(vote.eventType) == IAssessment.EventType.CLAIM
+        ? claims[vote.eventId].poll
+        : incidents[vote.eventId].poll;
+
+      totalReward = _getTotalRewardForEvent(
+        CONFIG,
+        IAssessment.EventType(vote.eventType),
+        vote.eventId,
+        claims,
+        incidents
+      );
+      rewardToWithdraw += totalReward * vote.tokenWeight / (poll.accepted + poll.denied);
+    }
+
+    stakeOf[user].rewardsWithdrawnUntilIndex = withdrawUntilIndex;
+    nxm.mint(user, rewardToWithdraw);
+  }
+
+  function castVote (
+    IAssessment.Configuration calldata CONFIG,
+    uint8 eventType,
+    uint104 id,
+    bool accepted,
+    mapping(address => IAssessment.Stake) storage stakeOf,
+    mapping(address => IAssessment.Vote[]) storage votesOf,
+    mapping(bytes32 => bool) storage hasAlreadyVotedOn,
+    IAssessment.Claim[] storage claims,
+    IAssessment.Incident[] storage incidents
+  ) external {
+
+    {
+      bytes32 voteHash = keccak256(abi.encodePacked(id, msg.sender, eventType));
+      require(!hasAlreadyVotedOn[voteHash], "Already voted");
+      hasAlreadyVotedOn[voteHash] = true;
+    }
+
+    IAssessment.Stake memory stake = stakeOf[msg.sender];
+    require(stake.amount > 0, "A stake is required to cast votes");
+
+    uint payoutImpact;
+    IAssessment.Poll memory poll;
+    uint32 blockTimestamp = uint32(block.timestamp);
+    if (IAssessment.EventType(eventType) == IAssessment.EventType.CLAIM) {
+      IAssessment.Claim memory claim = claims[id];
+      poll = claims[id].poll;
+      payoutImpact = AssessmentUtilsLib._getPayoutImpactOfClaim(claim.details);
+      require(blockTimestamp < poll.end, "Voting is closed");
+    } else {
```

Comment:
if it's not a claim then what is it? what goes down this path

Reply:
It could be made more explicit by expanding the else statement to:
else if (IAssessment.EventType(eventType) == IAssessment.EventType.INCIDENT) {...} else { revert }